File: backend-java/src/main/java/com/example/modules/payments/publicapi/PaymentsPublicApi.java
// ONLY interface other modules/apps may import from the payments module.
package com.example.modules.payments.publicapi;

import com.example.modules.payments.internal.PaymentService;
import java.util.Optional;

public class PaymentsPublicApi {
    private final PaymentService svc = new PaymentService();
    public PaymentService.Payment authorize(String paymentId, String orderId, long amountCents) { return svc.authorize(paymentId, orderId, amountCents); }
    public Optional<PaymentService.Payment> capture(String paymentId) { return svc.capture(paymentId); }
    public Optional<PaymentService.Payment> refund(String paymentId) { return svc.refund(paymentId); }
}

File: backend-csharp/modules/payments/PublicApi/PaymentsPublicApi.cs
// ONLY interface other modules/apps may import from the payments module.
using BackendCsharp.Modules.Payments.Internal;
namespace BackendCsharp.Modules.Payments.PublicApi;

public class PaymentsPublicApi
{
    private readonly PaymentService _svc = new();
    public Payment Authorize(string paymentId, string orderId, long amountCents) => _svc.Authorize(paymentId, orderId, amountCents);
    public Payment? Capture(string paymentId) => _svc.Capture(paymentId);
    public Payment? Refund(string paymentId) => _svc.Refund(paymentId);
}
